metric/system/process/process_windows.go

```diff
@@ -133,11 +133,8 @@ func FetchNumThreads(pid int) (int, error) {
 
 // FillPidMetrics is the windows implementation
 func FillPidMetrics(_ resolve.Resolver, pid int, state ProcState, _ func(string) bool) (ProcState, error) {
-	user, err := getProcCredName(pid)
-	if err != nil {
-		return state, fmt.Errorf("error fetching username: %w", err)
-	}
-	state.Username = user
+	user, _ := getProcCredName(pid)
+	state.Username = user // we cannot access process token for system-owned protected processes
 
 	ppid, _ := getParentPid(pid)
 	state.Ppid = opt.IntWith(ppid)
```

Comment:
So if we don't error and return, then we are going to call getParentPid, which I think would return 0 for the ppid (assuming this will fail like getProcCredName).  That seems like a problem, we don't really know if the ppid is 0 or not.

I guess the question I have is.  Is the state that is returned when getProcCredName fails accurate?

Reply:
> Is the state that is returned when getProcCredName fails accurate

Not really. The current flow of this function is:
- Get Process username, return error if failed
- Get Parent PID, ignore error if encountered
- Get memory metrics, return error if failed 
- Get CPU metrics, return error if failed

For protected processes, we fail at 1st step (we try to get the token for a protected process, which is not allowed). But, we can access memory and CPU metrics.

So, the state returned when getProcCredName fails is not accurate as we miss out on certain metrics later.

Reply:
I'm worried about when/if getParentPid fails.  If it fails, we get 0 for the ppid.  My guess is that getParentPid would always succeed when getProcCredName succeeded.  So this would be a new case where we are reporting the ppid incorrectly.

Reply:
@leehinman I've found a solution to this. Let me know how you feel about this.